src/wallet/wallet.ts

```diff
@@ -1099,19 +1099,43 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
       address: string | null,
       changeAddress: string | null,
       createMintAuthority: boolean,
+      mintAuthorityAddress: string | null,
+      allowExternalMintAuthorityAddress: boolean | null,
       createMeltAuthority: boolean,
+      meltAuthorityAddress: string | null,
+      allowExternalMeltAuthorityAddress: boolean | null,
       nftData: string | null,
       pinCode: string | null,
     };
     const newOptions: optionsType = Object.assign({
       address: null,
       changeAddress: null,
       createMintAuthority: true,
+      mintAuthorityAddress: null,
+      allowExternalMintAuthorityAddress: false,
       createMeltAuthority: true,
+      meltAuthorityAddress: null,
+      allowExternalMeltAuthorityAddress: false,
       nftData: null,
       pinCode: null,
     }, options);
 
+    if (newOptions.mintAuthorityAddress && !newOptions.allowExternalMintAuthorityAddress) {
+      // Validate that the mint authority address belongs to the wallet
+      const checkAddressMineMap = await this.checkAddressesMine([newOptions.mintAuthorityAddress]);
```

Comment:
Maybe we should implement `this.isAddressMine(...)` on the wallet-service facade

We can even use `HathorWalletServiceFacade.newAddresses` to avoid unnecessary calls to the lambda

Reply:
Yes, we should. We have an issue for that (https://github.com/HathorNetwork/hathor-wallet/issues/354) and it probably needs a design first because the old facade implements it in O(1) and for the wallet service it won't be possible to do it for all cases.